### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by MirkoCovizzi**
General cleanups:

* Name consistency

* Type consistency

* Mutability consistency

### CODE REVIEW

File: examples/again/src/lib.rs [hunk 1]
```diff
@@ -384,12 +384,12 @@ impl IAudioProcessor for AGainProcessor {
...
         }
     }
-    unsafe fn get_latency_sample(&self) -> u32 {
+    unsafe fn get_latency_samples(&self) -> u32 {
         info!("Called: AGainProcessor::get_latency_sample()");
 
         0
     }
-    unsafe fn setup_processing(&self, setup: *mut ProcessSetup) -> tresult {
+    unsafe fn setup_processing(&self, setup: *const ProcessSetup) -> tresult {
```
Comment (m-hilgendorf) [inline examples/again/src/lib.rs:392]: This is `ProcessSetup&` in the headers, should it be `const` here?

File: examples/again/src/lib.rs [hunk 2]
```diff
@@ -870,7 +873,7 @@ impl IUnitInfo for AGainController {
         &self,
         _list_or_unit: i32,
         _program_index: i32,
-        _data: *mut dyn IBStream,
+        _data: *mut c_void,
```
Comment (m-hilgendorf) [inline examples/again/src/lib.rs:876]: fixed in #28

File: src/base/ibstream.rs [hunk 3]
```diff
@@ -11,7 +11,7 @@ pub trait IBStream: IUnknown {
         -> tresult;
     unsafe fn write(
         &self,
-        buffer: *mut c_void,
+        buffer: *const c_void,
```
Comment (m-hilgendorf) [inline src/base/ibstream.rs:14]: this needs to be `mut` I think?

File: src/base/icloneable.rs [hunk 4]
```diff
@@ -1,6 +1,7 @@
+use std::os::raw::c_void;
 use vst3_com::{com_interface, interfaces::iunknown::IUnknown};
 
 #[com_interface("D45406B9-3A2D-4443-9DAD-9BA985A1454B")]
 pub trait ICloneable: IUnknown {
-    unsafe fn clone(&self) -> *mut dyn IUnknown;
+    unsafe fn clone(&self) -> *mut c_void;
```
Comment (m-hilgendorf) [inline src/base/icloneable.rs:6]: fixed in #28

File: src/base/ierrorcontext.rs [hunk 5]
```diff
@@ -1,10 +1,11 @@
...
+use crate::base::tresult;
+use std::os::raw::c_void;
 use vst3_com::com_interface;
 
 #[com_interface("12BCD07B-7C69-4336-B7DA-77C3444A0CD0")]
 pub trait IErrorContext: IUnknown {
     unsafe fn disable_error_ui(&self, state: bool);
     unsafe fn error_message_shown(&self) -> tresult;
-    unsafe fn get_error_message(&self, message: *mut dyn IString) -> tresult;
+    unsafe fn get_error_message(&self, message: *mut c_void) -> tresult;
```
Comment (m-hilgendorf) [inline src/base/ierrorcontext.rs:10]: fixed in #28

File: src/vst/ivstunits.rs [hunk 6]
```diff
@@ -1,94 +1,102 @@
...
+
+pub const kNoProgramListId: i32 = -1;
 
 #[repr(C)]
 #[derive(Copy, Clone)]
 pub struct UnitInfo {
     pub id: i32,
     pub parent_unit_id: i32,
-    pub name: String128,
+    pub name: [u16; 128],
```
Comment (m-hilgendorf) [inline src/vst/ivstunits.rs]: We should probably have a similar wrapper name around this to indicate that it's a widestring, but unsigned.

**Review by m-hilgendorf — CHANGES_REQUESTED** [review]
Just a few things, the `*dyn Interface` stuff was fixed in #28, but should go over it.

File: examples/again/src/lib.rs (hunk 1 above)
Comment (MirkoCovizzi) [reply examples/again/src/lib.rs:392]: Yes, as far as I have seen, it's always read only. It's a simple struct with basic values like sample rate. The method setup_processing gets called always before processing in the state machine and it's a way for the host to inform the plugin before the processing step.

File: src/base/ibstream.rs (hunk 3 above)
Comment (MirkoCovizzi) [reply src/base/ibstream.rs:14]: This one is a matter of taste. First of all, the naming of the IBStream methods are kinda backwards: write is for reading from the buffer (so *const would make sense), read is for writing to the buffer. You can see that for example in the again example in the SDK (and also in the again example in vst3-sys).

File: src/vst/ivstunits.rs (hunk 6 above)
Comment (MirkoCovizzi) [reply src/vst/ivstunits.rs]: That's fine, I can revert to String128.

Comment (MirkoCovizzi) [reply src/vst/ivstunits.rs]: Reverted

**Pull request merged**